### PR TITLE
3.x

### PR DESCRIPTION
**Pull request by mattcollier**
Depends on:

1. https://github.com/digitalbazaar/bedrock-key/pull/11
2. https://github.com/digitalbazaar/bedrock-passport/pull/25

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ function addRoutes(app) {
 
       // add public key
       brKey.addPublicKey(
-        req.user.identity, publicKey, privateKey, function(err) {
+        {actor: req.user.identity, publicKey, privateKey}, err => {
```
Comment (dlongley) [inline lib/index.js:75]: All of these will need to change to:

```{actor: req.user.actor, ...```

Once https://github.com/digitalbazaar/bedrock-passport/pull/24 is merged.

**Review by dlongley — CHANGES_REQUESTED** [review]
Will review more when I can, just wanted to note that we should base this against the `account` changes so as to reduce revs.

**Comment by dlongley** [comment]
Also depends on: https://github.com/digitalbazaar/bedrock-passport/pull/24

File: lib/index.js [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
+const cors = require('cors');
+const docs = require('bedrock-docs');
+const util = require('util');
 
-var logger = bedrock.loggers.get('app');
+const BedrockError = bedrock.util.BedrockError;
+const ensureAuthenticated = brPassport.ensureAuthenticated;
+const validate = require('bedrock-validation').validate;
+
+const logger = bedrock.loggers.get('app');
```
Comment (davidlehn) [inline lib/index.js:22]: Use a child logger.

**Pull request closed without merging**